tests/test_meshtastic_connection.py

```
@@ -307,3 +311,187 @@ def test_tcp_connection_returns_none_client_raises_connection_error(self):
+        assert mock_ble_ctor.call_count >= 1
+
+    def test_late_barrier_after_iface_creation_rolls_back_published_iface(self):
+        from concurrent.futures import Future
+
+        from mmrelay.meshtastic.connection import _connect_meshtastic_impl
+
+        class BleInterfaceWithConnect:
+            def __init__(
+                self,
+                address: str,
+                noProto: bool,  # noqa: N803
```

Ruff (0.15.10)

[warning] Boolean-typed positional argument in function definition

(FBT001)